src/java.base/share/classes/jdk/internal/foreign/SharedScope.java

```diff
@@ -45,6 +44,16 @@
 
     private static final ScopedMemoryAccess SCOPED_MEMORY_ACCESS = ScopedMemoryAccess.getScopedMemoryAccess();
 
+    private static final VarHandle STATE;
+
+    static {
+        try {
+            STATE = MethodHandles.lookup().findVarHandle(SharedScope.class, "state", int.class);
+        } catch (Throwable ex) {
+            throw new ExceptionInInitializerError(ex);
+        }
+    }
+
```

Comment:
This shouldn't be needed. It's already in ResourceScopeImpl

Reply:
That's right - removed.